### PR TITLE
Fix long group descriptions in NCD

### PR DESCRIPTION
**Pull request by sk7**

### CODE REVIEW

**Comment by sk7** [comment]
problem now: description only uses horizontal space that is available from the container. this means for small group names it will truncate the description in the story book very soon.

Screenshot 2021-04-20 at 12 22 38

 this is not the case in the NCD

File: web/src/stories/group-item/base.stories.tsx [hunk 1]
```diff
@@ -56,8 +56,9 @@ export const Type = (): JSX.Element => (
 )
 
 export const Truncated = (): JSX.Element => (
-  <Flex gap="2x" direction="column" style={{ width: 200 }} wrap>
+  <Flex gap="2x" direction="column" style={{ width: 500 }} width="100%" wrap>
```
Comment (sk7) [inline web/src/stories/group-item/base.stories.tsx:59]: this `style={{ width: 500 }} width="100%"` looks ugly, what do you think?

File: web/src/components/layout/flex/styles.tsx [hunk 2]
```diff
@@ -10,6 +10,7 @@ export default createUseStyles(() => {
     flex: {
       display: 'flex',
       flex: 1,
+      width: ({ width }) => width,
```
Comment (AleshaOleg) [inline web/src/components/layout/flex/styles.tsx:13]: let's use 100% as default here? 
`width: ({ width }) => width || '100%',`

File: web/src/stories/group-item/base.stories.tsx (hunk 1 above)
Comment (AleshaOleg) [reply web/src/stories/group-item/base.stories.tsx:59]: Let's remove `maxWidth` property from AvatarItem and GroupItem, then we will control their length via parent width property. That's why I think we need to keep this, or just create a parent div element and apply width there, to show that element will be limited to parent width

File: web/src/components/avatar/group-item/styles.ts [hunk 3]
```diff
@@ -34,6 +41,7 @@ export default createUseStyles(
     },
     membersCount: {
       marginLeft: theme.spaceMembersCount,
+      marginRight: theme.spaceMembersSecondary,
```
Comment (AleshaOleg) [inline web/src/components/avatar/group-item/styles.ts:44]: nice one, much easier then applying margin-left to desctiption class

File: web/src/stories/group-item/base.stories.tsx (hunk 1 above)
Comment (AleshaOleg) [reply web/src/stories/group-item/base.stories.tsx:59]: I added here width: 200, to show that truncating also works for name

File: web/src/components/avatar/group-item/index.tsx [hunk 4]
```diff
@@ -50,13 +50,13 @@ export const GroupItem: React.FC<GroupItemProps> = props => {
...
         type="button"
-        className={clsx(itemClasses.wrapper, className)}
+        className={clsx(itemClasses.wrapper, className, classes.button)}
         ref={ref}
         aria-label={ariaLabel || name}
         {...buttonProps}
       >
         <Group isUnclickable {...restGroupProps} />
-        <div className={itemClasses.text}>
+        <div className={clsx(itemClasses.text, classes.textWrapper)}>
```
Comment (AleshaOleg) [inline web/src/components/avatar/group-item/index.tsx:59]: Let's keep it consistent, either class name `text`, either `textWrapper`. Should be similar for both I guess

File: web/src/components/avatar/group-item/index.tsx [hunk 5]
```diff
@@ -50,13 +50,13 @@ export const GroupItem: React.FC<GroupItemProps> = props => {
     <FocusRing focusRingClass={onFocus} within>
       <button
         type="button"
-        className={clsx(itemClasses.wrapper, className)}
+        className={clsx(itemClasses.wrapper, className, classes.button)}
```
Comment (AleshaOleg) [inline web/src/components/avatar/group-item/index.tsx:53]: Same here, let's keep name same

**Review by AleshaOleg — CHANGES_REQUESTED** [review]

File: web/src/components/layout/flex/styles.tsx (hunk 2 above)
Comment (sk7) [reply web/src/components/layout/flex/styles.tsx:13]: I was afraid to do that but that would be even better :D

Comment (sk7) [reply web/src/components/layout/flex/styles.tsx:13]: isn't Flex used in a lot of places? we could break something

**Comment by sk7** [comment]
@AleshaOleg weirdly enough the name gets also truncated with 500px width

Screenshot 2021-04-22 at 11 51 07

**Comment by AleshaOleg** [comment]
image

in branch `bug/group-long-description/GRAPE-19558-2` works fine with width: 500px

File: web/src/components/layout/flex/styles.tsx (hunk 2 above)
Comment (AleshaOleg) [reply web/src/components/layout/flex/styles.tsx:13]: Most likely it will not as wrapper parent exist for each Flex component

**Comment by sk7** [comment]
ok merged your changes here and fixed the class names 👍

**Comment by AleshaOleg** [comment]
lgtm, merging :)

**Pull request merged**